pkg/policy/api/rule_validation.go

```diff
@@ -81,8 +98,29 @@ func (i *IngressRule) sanitize() error {
 }
 
 func (e *EgressRule) sanitize() error {
-	if len(e.ToCIDR) > 0 && len(e.ToPorts) > 0 {
-		return fmt.Errorf("Combining ToPorts and ToCIDR is not supported yet")
+	l3Members := map[string]int{
+		"ToCIDR":     len(e.ToCIDR),
+		"ToCIDRSet":  len(e.ToCIDRSet),
+		"ToEntities": len(e.ToEntities),
+		"ToServices": len(e.ToServices),
+	}
+	l3DependentL4Support := map[interface{}]bool{
+		"ToCIDR":     false,
+		"ToCIDRSet":  false,
+		"ToEntities": false,
+		"ToServices": false,
+	}
+	for m1 := range l3Members {
+		for m2 := range l3Members {
+			if m2 != m1 && l3Members[m1] > 0 && l3Members[m2] > 0 {
```

Comment:
This will cause ToServices to break, as ToServices rule is being translated into ToCIDRSet rule. This code needs a bit more logic that handles that case.

Reply:
Cool. Rather than opening a hole for configuration of things that violate the general rule semantics, I've attempted to reshuffle the sanitization logic so that internally we can represent things in whichever way makes sense, but externally we should only advertise support for particular combinations. This approach is represented by the new patch "daemon: Shift policy sanitize up to handle".